### PR TITLE
Copter: fix AHRS and VisualOdom pre-arm attitude consistency checks

### PR DESCRIPTION
**Pull request by rmackay9**
This PR fixes two similar pre-arm attitude consistency checks, one in AP_AHRS and another in AP_VisualOdom.

The issue was that the roll, pitch angle comparison was failing and alerting the user when in fact, it was the yaw that was different. This was especially troublesome in cases where we only want to compare roll and pitch (and not yaw) for example when using external nav for yaw across DCM and EKF3.

This has been tested in SITL and on a real board.

Resolves issue https://github.com/ArduPilot/ardupilot/issues/17442

### CODE REVIEW

File: libraries/AP_Math/quaternion.cpp [hunk 1]
```diff
@@ -692,3 +692,22 @@ Quaternion Quaternion::angular_difference(const Quaternion &v) const
 {
     return v.inverse() * *this;
 }
+
+// calculate earth-frame roll-pitch difference (in radians) between this Quaternion and another
```
Comment (tridge) [inline libraries/AP_Math/quaternion.cpp:696]: returns abs value in comment

File: libraries/AP_Math/quaternion.cpp [hunk 2]
```diff
@@ -692,3 +692,22 @@ Quaternion Quaternion::angular_difference(const Quaternion &v) const
...
+    Matrix3f m, vm;
+    rotation_matrix(m);
+    v.rotation_matrix(vm);
+
+    // rotate earth frame vertical vector by each rotation matrix
+    const Vector3f z_unit_vec{0,0,1};
+    const Vector3f z_unit_m = m.mul_transpose(z_unit_vec);
+    const Vector3f z_unit_vm = vm.mul_transpose(z_unit_vec);
+    const Vector3f vec_diff = z_unit_vm - z_unit_m;
+    const float vec_len_div2 = constrain_float(vec_diff.length() * 0.5f, -1.0f, 1.0f);
```
Comment (tridge) [inline libraries/AP_Math/quaternion.cpp:709]: constrain 0 to 1?

**Review by tridge — DISMISSED** [review]

File: libraries/AP_Math/quaternion.cpp (hunk 2 above)
Comment (rmackay9) [reply libraries/AP_Math/quaternion.cpp]: fixed

File: libraries/AP_Math/quaternion.cpp (hunk 1 above)
Comment (rmackay9) [reply libraries/AP_Math/quaternion.cpp]: changed comment, txs

**Pull request merged**